--- origsrc/gnome-keyring-2.30.3/tests/gtest.make	2010-06-21 08:51:39.000000000 -0500
+++ src/gnome-keyring-2.30.3/tests/gtest.make	2010-06-28 01:16:36.798436100 -0500
@@ -26,7 +26,7 @@ noinst_PROGRAMS= \
 	run-prompt-test
 		
 run-auto-test.h: $(UNIT_AUTO) Makefile.am $(top_srcdir)/tests/prep-gtest.sh
-	sh $(top_srcdir)/tests/prep-gtest.sh -b run-auto-test $(UNIT_AUTO)
+	sh $(top_srcdir)/tests/prep-gtest.sh -b run-auto-test $(addprefix $(srcdir)/,$(UNIT_AUTO))
 
 run-auto-test.c: run-auto-test.h
 
@@ -42,7 +42,7 @@ run_auto_test_CFLAGS = \
 	$(UNIT_FLAGS)
 
 run-prompt-test.h: $(UNIT_PROMPT) Makefile.am $(top_srcdir)/tests/prep-gtest.sh
-	sh $(top_srcdir)/tests/prep-gtest.sh -b run-prompt-test $(UNIT_PROMPT)
+	sh $(top_srcdir)/tests/prep-gtest.sh -b run-prompt-test $(addprefix $(srcdir)/,$(UNIT_PROMPT))
 
 run-prompt-test.c: run-prompt-test.h
 
