--- origsrc/gnome-keyring-2.32.1/egg/egg-asn1x.c	2010-10-18 21:11:38.000000000 -0500
+++ src/gnome-keyring-2.32.1/egg/egg-asn1x.c	2010-11-16 13:56:02.893901000 -0600
@@ -1988,8 +1988,8 @@ anode_write_integer_ulong (gulong value,
 	guchar buf[sizeof (gulong)];
 	gint bytes, i, off;
 
-	for (i = 0; i < sizeof (ulong); ++i) {
-		off = sizeof (ulong) - (i + 1);
+	for (i = 0; i < sizeof (unsigned long); ++i) {
+		off = sizeof (unsigned long) - (i + 1);
 		buf[i] = (value >> (off * 8)) & 0xFF;
 	}
 
