--- origsrc/gnome-keyring-2.32.1/gcr/Makefile.am	2010-10-25 08:59:09.000000000 -0500
+++ src/gnome-keyring-2.32.1/gcr/Makefile.am	2010-11-16 13:47:18.713289100 -0600
@@ -59,7 +59,7 @@ libgcr@GCR_VERSION_SUFFIX@_la_SOURCES =
 	$(BUILT_SOURCES)
 
 libgcr@GCR_VERSION_SUFFIX@_la_CFLAGS = \
-	-DPKCS11_MODULE_PATH=\""$(libdir)/gnome-keyring/gnome-keyring-pkcs11.so"\" \
+	-DPKCS11_MODULE_DIR=\""$(libdir)/gnome-keyring/"\" \
 	-DGCR_API_SUBJECT_TO_CHANGE \
 	-DUIDIR=\""$(uidir)"\"
     
--- origsrc/gnome-keyring-2.32.1/gcr/gcr-library.c	2010-10-25 08:59:09.000000000 -0500
+++ src/gnome-keyring-2.32.1/gcr/gcr-library.c	2010-11-16 13:47:53.843794500 -0600
@@ -121,7 +121,7 @@ _gcr_initialize (void)
 	if (g_once_init_enter (&gcr_initialized)) {
 		
 		/* TODO: This needs reworking for multiple modules */
-		module = gp11_module_initialize (PKCS11_MODULE_PATH, NULL, &error);
+		module = gp11_module_initialize (PKCS11_MODULE_DIR "gnome-keyring-pkcs11." G_MODULE_SUFFIX, NULL, &error);
 		if (module) {
 			gp11_module_set_pool_sessions (module, TRUE);
 			gp11_module_set_auto_authenticate (module, TRUE);
