--- origsrc/gnome-keyring-2.32.1/gp11/tests/Makefile.am	2010-10-25 08:59:10.000000000 -0500
+++ src/gnome-keyring-2.32.1/gp11/tests/Makefile.am	2010-11-16 13:46:29.796439900 -0600
@@ -28,7 +28,7 @@ include $(top_srcdir)/testing/testing.ma
 lib_LTLIBRARIES = libgp11-test-module.la
 
 libgp11_test_module_la_LDFLAGS = \
-	-avoid-version
+	-module -avoid-version -no-undefined
 
 libgp11_test_module_la_CFLAGS = \
 	-I$(top_srcdir)/gp11 \
--- origsrc/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-crypto.c	2010-10-25 08:59:10.000000000 -0500
+++ src/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-crypto.c	2010-11-16 13:45:03.854584600 -0600
@@ -18,7 +18,7 @@ DEFINE_SETUP(crypto_session)
 	GList *slots;
 	
 	/* Successful load */
-	module = gp11_module_initialize (".libs/libgp11-test-module.so", NULL, &err);
+	module = gp11_module_initialize (".libs/cyggp11-test-module.dll", NULL, &err);
 	SUCCESS_RES (module, err);
 	
 	slots = gp11_module_get_slots (module, TRUE);
--- origsrc/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-module.c	2010-10-25 08:59:10.000000000 -0500
+++ src/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-module.c	2010-11-16 13:45:45.504041100 -0600
@@ -12,7 +12,7 @@ DEFINE_SETUP(load_module)
 	GError *err = NULL;
 
 	/* Successful load */
-	module = gp11_module_initialize (".libs/libgp11-test-module.so", NULL, &err);
+	module = gp11_module_initialize (".libs/cyggp11-test-module.dll", NULL, &err);
 	SUCCESS_RES (module, err);
 }
 
@@ -64,7 +64,7 @@ DEFINE_TEST(module_props)
 
 	g_object_get (module, "path", &path, NULL);
 	g_assert (path != NULL && "no module-path");
-	g_assert (strcmp (".libs/libgp11-test-module.so", path) == 0 && "module path wrong");
+	g_assert (strcmp (".libs/cyggp11-test-module.dll", path) == 0 && "module path wrong");
 	g_free (path);
 }
 
--- origsrc/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-object.c	2010-10-25 08:59:10.000000000 -0500
+++ src/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-object.c	2010-11-16 13:45:54.206910900 -0600
@@ -19,7 +19,7 @@ DEFINE_SETUP(prep_object)
 	GList *slots;
 	
 	/* Successful load */
-	module = gp11_module_initialize (".libs/libgp11-test-module.so", NULL, &err);
+	module = gp11_module_initialize (".libs/cyggp11-test-module.dll", NULL, &err);
 	SUCCESS_RES (module, err);
 	
 	slots = gp11_module_get_slots (module, TRUE);
--- origsrc/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-session.c	2010-10-25 08:59:10.000000000 -0500
+++ src/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-session.c	2010-11-16 13:46:02.055937100 -0600
@@ -18,7 +18,7 @@ DEFINE_SETUP(load_session)
 	GList *slots;
 	
 	/* Successful load */
-	module = gp11_module_initialize (".libs/libgp11-test-module.so", NULL, &err);
+	module = gp11_module_initialize (".libs/cyggp11-test-module.dll", NULL, &err);
 	SUCCESS_RES (module, err);
 	
 	slots = gp11_module_get_slots (module, TRUE);
--- origsrc/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-slot.c	2010-10-25 08:59:10.000000000 -0500
+++ src/gnome-keyring-2.32.1/gp11/tests/unit-test-gp11-slot.c	2010-11-16 13:46:09.457736700 -0600
@@ -14,7 +14,7 @@ DEFINE_SETUP(load_slots)
 	GList *slots;
 	
 	/* Successful load */
-	module = gp11_module_initialize (".libs/libgp11-test-module.so", NULL, &err);
+	module = gp11_module_initialize (".libs/cyggp11-test-module.dll", NULL, &err);
 	SUCCESS_RES (module, err);
 	
 	slots = gp11_module_get_slots (module, TRUE);
