--- origsrc/gnome-keyring-3.0.1/gck/tests/Makefile.am	2011-04-24 02:59:30.000000000 -0500
+++ src/gnome-keyring-3.0.1/gck/tests/Makefile.am	2011-04-29 19:35:50.349302500 -0500
@@ -36,7 +36,7 @@ check-local: test
 
 all-local: $(check_PROGRAMS)
 
-lib_LTLIBRARIES = libmock-test-module.la
+noinst_LTLIBRARIES = libmock-test-module.la
 
 libmock_test_module_la_LDFLAGS = \
 	-avoid-version
--- origsrc/gnome-keyring-3.0.1/pkcs11/wrap-layer/tests/Makefile.am	2011-04-24 03:03:14.000000000 -0500
+++ src/gnome-keyring-3.0.1/pkcs11/wrap-layer/tests/Makefile.am	2011-04-29 16:33:12.278182100 -0500
@@ -4,7 +4,8 @@ INCLUDES = \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/pkcs11 \
 	-DSRCDIR="\"$(srcdir)\"" \
-	$(GLIB_CFLAGS) \
+	$(GOBJECT_CFLAGS) \
+	$(LIBGCRYPT_CFLAGS) \
 	-DWITH_TESTABLE
 
 LDADD = \
@@ -13,7 +14,8 @@ LDADD = \
 	$(top_builddir)/pkcs11/gkm/libgkm.la \
 	$(top_builddir)/ui/libgku-prompt-testable.la \
 	$(top_builddir)/egg/libegg.la \
-	$(GLIB_LIBS)
+	$(GOBJECT_LIBS) \
+	$(LIBGCRYPT_LIBS)
 
 TEST_PROGS = \
 	test-create-credential \
