--- origsrc/gcr-3.4.1/configure.ac	2012-04-16 06:54:07.000000000 -0500
+++ src/gcr-3.4.1/configure.ac	2012-05-04 00:56:38.715304600 -0500
@@ -91,6 +91,7 @@ AC_SUBST(GTK_LIBS)
 
 AC_CHECK_FUNCS(timegm, AC_DEFINE(HAVE_TIMEGM,1,[Have timegm]))
 AC_CHECK_FUNCS(mlock)
+AC_CHECK_FUNCS(memrchr)
 
 # --------------------------------------------------------------------
 # p11-kit
--- origsrc/gcr-3.4.1/egg/egg-armor.c	2011-11-23 11:14:05.000000000 -0600
+++ src/gcr-3.4.1/egg/egg-armor.c	2012-05-04 03:54:37.391089900 -0500
@@ -34,6 +34,26 @@
 #include <ctype.h>
 #include <string.h>
 
+#ifndef HAVE_MEMRCHR
+void *memrchr (const void *, int, size_t);
+
+void *
+memrchr (const void *src, int c, size_t len)
+{
+	const void *src_end = (const unsigned char *)src + len - 1;
+	const void *last = NULL;
+
+	while ((src = memchr (src, c, len))) {
+		if (src > src_end)
+			break;
+		last = src;
+		src = (const unsigned char *)src + 1;
+	}
+
+	return (void *) last;
+}
+#endif
+
 /*
  * Armor looks like:
  *
